test/wasi/c/symlink_loop.c

```diff
@@ -0,0 +1,9 @@
+#include <assert.h>
+#include <errno.h>
+#include <stdio.h>
+
+int main() {
+  FILE* file = fopen("/sandbox/subdir/loop1", "r");
+  assert(file == NULL);
+  assert(errno == ELOOP);
```

Comment:
@cjihrig would it be possible to create these recursive symlinks dynamically during the test?

The error that is occurring above, ELOOP, is also being found by tooling that recurses the files from a nodejs/node checkout. I'm working around it, but it broke part of my work flow:
```
$ git ls-files -z $(git rev-parse --show-cdup) | tr '\0' '\n' | entr -crx ninja -C out/Release
entr: cannot stat 'test/fixtures/wasi/subdir/loop1': Too many levels of symbolic link
```

Reply:
@sam-github I'll send a PR later this evening.